### ISSUE TITLE
Quotation style enforcement?

### ISSUE THREAD

**Issue by awayken** [rossPatton/stylint#47]
A feature that I like from JSHint is the ability to enforce quotation types. I try to enforce double quotes in my CSS/Stylus and single quotes in my JavaScript. Is that something others would find valuable in Stylint?

That said, it looks like JSHint is moving away from enforcing this rule altogether: http://jshint.com/docs/options/#quotmark

**Comment by rossPatton** [comment]
Yeah, it looks like jshint is letting jscs handle issues of code style in the future.

Personally, I see stylint as being a linter for both code correctness and code style, so a rule for enforcing quotation marks makes sense to me.

**Comment by rossPatton** [comment]
if you pull the latest, this feature should be available now, could use some testing

**Comment by awayken** [comment]
Leave it to me to find some edge cases. :wink: 

I found some inconsistency when checking quotes in selectors. Hopefully the example below makes the issue clear.

quotes.styl
```stylus
// Passes but shouldn't
[class*="--button"] {
    border: 1px solid $primary-color;
}

// Last line passes but shouldn't
// Other lines fail as expected
[class*="--button"],
[class*="--bigbutton"],
input[type="text"],
input[type="button"] {
    border: 1px solid $primary-color;
}
```

quotes.json
```json
{
    "alphabetical": true,
    "borderNone": true,
    "brackets": false,
    "colons": false,
    "colors": false,
    "commaSpace": true,
    "commentSpace": false,
    "cssLiteral": false,
    "depthLimit": false,
    "duplicates": true,
    "efficient": true,
    "enforceBlockStyle": false,
    "enforceVarStyle": false,
    "extendPref": false,
    "globalDupe": false,
    "indentSpaces": 4,
    "leadingZero": true,
    "maxWarnings": 10,
    "maxWarningsKill": false,
    "mixed": false,
    "namingConvention": false,
    "parenSpace": false,
    "placeholders": true,
    "quotePref": "single",
    "semicolons": false,
    "universal": true,
    "valid": false,
    "whitespace": true,
    "zeroUnits": true,
    "zIndexDuplicates": false,
    "zIndexNormalize": false
}
```

Output:
```bash
Warning:  preferred quote style is single quotes
File: quotes.styl
Line: 14: [class*="--button"],

Warning:  preferred quote style is single quotes
File: quotes.styl
Line: 15: [class*="--bigbutton"],

Warning:  preferred quote style is single quotes
File: quotes.styl
Line: 16: input[type="text"],
```

**Comment by awayken** [comment]
There seems to be a problem checking for quote preference when setting argument defaults. If I specify `quotePref: "single"`, the following still passes:

```stylus
show-content( $content = "Hello!" ) {
    &::before {
        content: $content;
    }
}
```

**Comment by rossPatton** [comment]
interesting that you're hitting these issues - i've test all of the cases here with all the quotePref settings and they all appear to be behaving as expected, ie, i can't replicate the issue

someone else want to test this?

**Comment by awayken** [comment]
@rossPatton I'm testing with version `Stylint version:  0.9.2`. If you create files and use the settings as I have them above, do you see the issue like I do?

**Comment by rossPatton** [comment]
Yeah, i always test by running stylint against real stylus files, and then again with unit tests.

In both cases I didn't have this issue.

I'll take another look, and make sure i'm testing with 0.9.2 instead of the latest on my local, and post an update here.

**Comment by jackbrewer** [comment]
I'm using `master` and am seeing the same output as pasted above: only 3 warnings reported for the file, even though 5 exist.

I noticed if I add additional classes to the selectors, I can get all 5 to report warnings as expected.

```
[class*="--button"],
.additional-class {
    border: 1px solid $primary-color;
}

[class*="--button"],
[class*="--bigbutton"],
input[type="text"],
input[type="button"],
.another-additional-class {
    border: 1px solid $primary-color;
}
```

Also, when using my preferred minimal syntax, all 5 warnings are reported as expected, without needing the additional classes:
```
[class*="--button"]
  border 1px solid $primary-color

[class*="--button"]
[class*="--bigbutton"]
input[type="text"]
input[type="button"]
  border 1px solid $primary-color
```

Maybe something to do with attribute selectors next to curly braces?

**Comment by rossPatton** [comment]
@jackbrewer i can take another look, thanks for providing more examples.

could you try pulling develop and testing against that as well?

**Comment by jackbrewer** [comment]
Same output as master – 3 warning for the original stylus, full 5 warnings with the modifications from my above message. Tested on latest `develop` (d3a01bc)

**Comment by rossPatton** [comment]
Thanks!

**Comment by awayken** [comment]
Not sure if this is helpful, but I pulled the latest develop (31638855f57d7c5d1b7b870d5ca274f9d0dbe181), and I still see issues.

**Comment by rossPatton** [comment]
@awayken yeaaah, that's to be expected, i haven't really gotten to this one yet. soon though, i intend this fix to be in 0.9.5

**Comment by rossPatton** [comment]
@awayken aaand this should be fixed now (in develop). Lemme know!

**Comment by awayken** [comment]
The only case that still doesn't pass is in the function declaration.

```stylus
show-content( $content = "Hello!" ) {
    &::before {
        content: $content;
    }
}
```

I'm using fa1ee81ad674eda65f8cdde1cfef3da4e1802428.

**Comment by rossPatton** [comment]
@awayken should be good now. it's now fixed in 0.9.5

**Comment by awayken** [comment]
:+1: Looks good to me. Just tested with 26e58b76464e297d2a2c81f16e0d2823ca1e1777.
